### PR TITLE
[AzureStack] Update Fabric Admin version to 2018-10-01

### PR DESCRIPTION
**Pull request by zyxyoshine**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

API spec PR: https://github.com/Azure/azure-rest-api-specs/pull/4255
### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by zyxyoshine** [comment]
@bganapa   @deathly809

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/DriveTests.cs [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+
+        [Fact]
+        public void TestListDrives()
+        {
+            RunTest((client) => {
+                OverStorageSubSystems(client, (fabricLocationName, scaleUnitName, storageSubSystemName) => {
+                    var drives = client.Drives.List(ResourceGroupName, fabricLocationName, scaleUnitName, storageSubSystemName);
+                    Common.MapOverIPage(drives, client.Drives.ListNext, ValidateDrive);
+                    Common.WriteIPagesToFile(drives, client.Drives.ListNext, "ListDrives.txt", ResourceName);
```
Comment (deathly809) [inline src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/DriveTests.cs]: This method was really only for debugging purposes.  Mainly so I could visually verify everything was okay.  I should do a PR to remove these one day ha ha.

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/DriveTests.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (deathly809) [inline src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/DriveTests.cs]: You should have a negative test, to make sure when you try to get an invalid drive it returns 404 like expected.  Also, if there are drives that exist but should never be retrieved you should add a test for that too!

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/StorageSubSystemTests.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (deathly809) [inline src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/StorageSubSystemTests.cs]: Add a test for invalid storage subsystems.

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/VolumeTests.cs [hunk 4]
```diff
@@ -21,29 +21,36 @@ public class VolumeTests : FabricTestBase
             {
```
Comment (deathly809) [inline src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/VolumeTests.cs]: Add a test for invalid volumes.

**Review by deathly809 — CHANGES_REQUESTED** [review]
Overall looks good!  Just add some tests for invalid input and maybe test to see if you can grab objects (if any exist) which you should not be able to get.  Also, did you create a tenant VM for the Restart/Shutdown/Start Tenant tests?  This is to ensure we fail to touch tenant resources.

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj [hunk 5]
```diff
@@ -7,7 +7,7 @@
     <PropertyGroup>
         <PackageId>Microsoft.AzureStack.Management.Fabric.Admin</PackageId>
         <Description>Provides developers with libraries for the fabric admin under AzureStack to manage fabric infrastructure.</Description>
-        <Version>0.3.0-preview</Version>
+        <Version>0.4.0-preview</Version>
         <AssemblyName>Microsoft.AzureStack.Management.Fabric.Admin</AssemblyName>
         <PackageTags>Microsoft Azure Stack;Fabric;REST HTTP client;azureostackfficial;netcore452</PackageTags>
         <PackageReleaseNotes>
```
Comment (dsgouda) [inline src/AzureStack/Admin/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj]: Please update `PackageReleaseNotes`
Also, please set   `<TreatWarningsAsErrors>true</TreatWarningsAsErrors>` and fix all errors reported

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin/generate.ps1 [hunk 6]
```diff
@@ -1 +1 @@
-Start-AutoRestCodeGeneration -ResourceProvider "azsadmin/resource-manager/fabric" -AutoRestVersion "latest" -SdkRootDirectory $PSScriptRoot
\ No newline at end of file
+Start-AutoRestCodeGeneration -ResourceProvider "azsadmin/resource-manager/fabric" -AutoRestVersion "latest" -SdkRootDirectory $PSScriptRoot -ConfigFileTag "package-2018-10-01"
```
Comment (dsgouda) [inline src/AzureStack/Admin/FabricAdmin/Fabric.Admin/generate.ps1]: Is there a reason why a specific tag is being passed here? Why is the code not being generated from a default tag?

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj [hunk 7]
```diff
@@ -7,7 +7,7 @@
     <PropertyGroup>
         <PackageId>Microsoft.AzureStack.Management.Fabric.Admin</PackageId>
         <Description>Provides developers with libraries for the fabric admin under AzureStack to manage fabric infrastructure.</Description>
-        <Version>0.3.0-preview</Version>
+        <Version>0.4.0-preview</Version>
         <AssemblyName>Microsoft.AzureStack.Management.Fabric.Admin</AssemblyName>
         <PackageTags>Microsoft Azure Stack;Fabric;REST HTTP client;azureostackfficial;netcore452</PackageTags>
```
Comment (bganapa) [inline src/AzureStack/Admin/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj]: azureostackfficial -> azurestack

i dont think official is needed here

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin/generate.ps1 (hunk 6 above)
Comment (zyxyoshine) [reply src/AzureStack/Admin/FabricAdmin/Fabric.Admin/generate.ps1]: Our API spec doesn't change the default tag to the new version, we decided to do it after release, thus, we need to specific tag for generate new SDK.

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj (hunk 5 above)
Comment (zyxyoshine) [reply src/AzureStack/Admin/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj]: Fixed.

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj (hunk 7 above)
Comment (zyxyoshine) [reply src/AzureStack/Admin/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj]: Fixed.

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/DriveTests.cs (hunk 2 above)
Comment (zyxyoshine) [reply src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/DriveTests.cs]: I have added some negative tests to test invalid value for Drive, StorageSubSystem and Volume.

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/StorageSubSystemTests.cs (hunk 3 above)
Comment (zyxyoshine) [reply src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/StorageSubSystemTests.cs]: Added.

File: src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/VolumeTests.cs (hunk 4 above)
Comment (zyxyoshine) [reply src/AzureStack/Admin/FabricAdmin/Fabric.Admin.Tests/src/VolumeTests.cs]: Added.

**Comment by bganapa** [comment]
I do  not think we plan any spec changes for the older api version 2016-05-01 or release another SDK for that api version .. @zyxyoshine please confirm

**Comment by bganapa** [comment]
@zyxyoshine we had a conversation with the SDK team, since we are moving up the API Version , the SDK major version needs to be updated. Still we consider 2018-10-01 as preview in the specs, Can you please update the SDK package version to be 1.8.0-preview, AssemblyVersion to be 1.0.0.0 and AssemblyFileVersion to be 1.8.0.0

**Comment by zyxyoshine** [comment]
@dsgouda @bganapa the version has been updated, can we merge the PR?

**Review by dsgouda — APPROVED** [review]

**Comment by dsgouda** [comment]
LGTM

**Pull request merged**